File: 0844-backspace-string-compare/0844-backspace-string-compare.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) { 
        return buildString(s) == buildString(t);
    }
    string buildString(string s) {
        string ans = "";
        for (char c : s) {
            if (c != '#') {
                ans.push_back(c);
            }
            else if (ans.empty() == false) {
                ans.pop_back();
            }
        }
        return ans;
    }
};

File: 0844-backspace-string-compare/0844-backspace-string-compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        def fix(r):
            stack = []
            for char in r:
                if char == "#":
                    if stack:
                        stack.pop()
                else:
                    stack.append(char)
            return "".join(stack)
        return fix(s) == fix(t)

        